File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        List<Integer> [] frequency = new List[nums.length + 1];
        Map<Integer, Integer> count = new HashMap<>();

        for(int i = 0; i < frequency.length; i++){
            frequency[i] = new ArrayList<>();
        }

        for(int i = 0; i < nums.length; i++){
            count.put(nums[i], count.getOrDefault(nums[i], 0) + 1);
        }


        for(Map.Entry<Integer, Integer> current: count.entrySet()){
            //System.out.println("value:" + current.getValue() + " Key:" + current.getKey());
            frequency[current.getValue()].add(current.getKey());
        }

        int index = 0;
        int [] result = new int [k];

        for(int i = frequency.length - 1; i > 0; i--){
           
            for(int n: frequency[i]){
                result[index++] = n;
                if(index == k){
                    return result;
                }
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/subsets/submission-10.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = []
        self.backtrack(nums, [], result, 0)
        return result

    def backtrack(self, nums, current, result, i):
        if i >= len(nums):
            result.append(current.copy())
            return
        
        current.append(nums[i])
        self.backtrack(nums, current, result, i + 1)

        current.pop()
        self.backtrack(nums, current, result, i + 1)
        
    
    




        